tests/envs/test_trafficjunction.py

```diff
@@ -0,0 +1,151 @@
+        assert env.observation_space.contains(obs), 'Observation must be part of the observation space'
+    assert env.observation_space.contains(obs), 'Observation must be part of the observation space'
+    assert env.observation_space.contains(env.observation_space.sample()), 'Observation must be part of the' \
+                                                                           ' observation space'
+
+
+@pytest_parametrize_plus('env',
+                         [fixture_ref(env_4)])
+def test_step_cost_env4(env):
+    env.reset()
+    for step_i in range(3):  # small number of steps so that no collision occurs
+        obs, reward_n, done, _ = env.step(env.action_space.sample())
+        target_reward = [env._step_cost * (step_i + 1) for _ in range(env.n_agents)]
+        assert (reward_n == target_reward), \
+            'step_cost is not correct. Expected {} ; Got {}'.format(target_reward, reward_n)
+
+
+@pytest_parametrize_plus('env',
+                         [fixture_ref(env_10)])
+def test_step_cost_env10(env):
+    env.reset()
+    for step_i in range(1):  # just 1 step so that no collision occurs
+        obs, reward_n, done, _ = env.step(env.action_space.sample())
+        target_reward = [env._step_cost * (step_i + 1) for _ in range(4)] + [0 for _ in range(env.n_agents - 4)]
+        assert (reward_n == target_reward), \
+            'step_cost is not correct. Expected {} ; Got {}'.format(target_reward, reward_n)
+
+
+@pytest_parametrize_plus('env',
+                         [fixture_ref(env_4)])
+def test_all_brake_rollout_env4(env):
+    """ All agent apply brake for the entire duration"""
+    for _ in range(2):
+        env.reset()
+        step_i = 0
+        done = [False for _ in range(env.n_agents)]
+        while not all(done):  # small number of steps so that no collision occurs
+            obs, reward_n, done, _ = env.step([1 for _ in range(env.n_agents)])
+            target_reward = [env._step_cost * (step_i + 1) for _ in range(env.n_agents)]
+            step_i += 1
+            assert (reward_n == target_reward), \
+                'step_cost is not correct. Expected {} ; Got {}'.format(target_reward, reward_n)
+        assert step_i == env._max_steps, 'max-steps should be reached'
+
+
+@pytest_parametrize_plus('env',
+                         [fixture_ref(env_4)])
+def test_one_gas_others_brake_rollout_env4(env):
+    """
+    "Agent 0" applies gas and others brake. This will mean that their will not be any collision and "Agent 0" will
+    reach it's destination in minimal number of steps; beyond which reward for agent "0" would be 0.
+    """
+
+    # testing over multiple episode to ensure it works with multiple routes assigned to "agent 0"
+    for episode_i in range(5):
+        obs = env.reset()
+        step_i = 0
+        done = [False for _ in range(env.n_agents)]
+        agent_0_route = obs[0].reshape((9, 9))[4][6:]  # one-hot
+        # Todo: Find max. steps for agent 0 based on it's route
```

Comment:
Hi @rafaelmp2, Thanks for the work so far. We are almost done and just need to add some test cases involving rollouts for sanity.

In this particular test case, we need to ensure that once the agent reaches the destination; it will get "0" reward for rest of the episode.

I am not sure of the "minimal number of steps" to reach the destination required for a agent when there is no collision and agent has applied "gas" at all time-steps. This number varies based on the route taken by the agent. 
It will be helpful if you can finish this test-case by adding the current number of steps. Currently, it has wrong values,.

Reply:
It will also help if you can add more rollout test cases, such as a scenario where some number of agents collide so that we can ensure that it gets the correct collision reward.